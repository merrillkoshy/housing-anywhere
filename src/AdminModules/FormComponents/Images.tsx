import React, { useState, useCallback } from "react";
import { useDropzone } from "react-dropzone";
import { Form, Button } from "react-bootstrap";
import { styles } from "./image-dropzone-styles";

const Images = ({
	files,
	setFiles,
}: {
	files: any;
	setFiles: React.Dispatch<React.SetStateAction<any>>;
}) => {
	const onDrop = useCallback((acceptedFiles) => {
		setFiles(
			acceptedFiles.map((file: any) =>
				Object.assign(file, {
					preview: URL.createObjectURL(file),
				})
			)
		);
	}, []);
	const { getRootProps, getInputProps, isDragActive } = useDropzone({ onDrop });
	const thumbs = files?.map((file: any) => (
		<div style={styles.thumb} key={file.name}>
			<div style={styles.thumbInner}>
				<img src={file.preview} style={styles.img} alt={"img-preview"} />
			</div>
		</div>
	));
	return (
		<>
			<div {...getRootProps()}>
				<input id="image-dropzone" {...getInputProps()} />
				{isDragActive ? (
					<p>Drop the files here ...</p>
				) : (
					<p>Drag 'n' drop some files here, or click to select files</p>
				)}
				<aside style={styles.thumbsContainer}>{thumbs}</aside>
			</div>
		</>
	);
};
export default Images;
