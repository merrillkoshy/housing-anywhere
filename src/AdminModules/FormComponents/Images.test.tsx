import React from "react";
import { shallow } from "enzyme";
import Images from "./Images";

describe("ImagesModule Costs module", () => {
	const ImagesModule = shallow(<Images />);
	it("renders", () => {
		expect(ImagesModule).toMatchSnapshot();
	});
	it("should have a dropzone for images", () => {
		expect(ImagesModule.find("#image-dropzone").exists()).toBe(true);
	});
});
