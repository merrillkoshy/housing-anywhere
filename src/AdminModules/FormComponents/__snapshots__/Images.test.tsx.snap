// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagesModule Costs module renders 1`] = `
<Fragment>
  <div
    onBlur={[Function]}
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
  >
    <input
      autoComplete="off"
      id="image-dropzone"
      multiple={true}
      onChange={[Function]}
      onClick={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      tabIndex={-1}
      type="file"
    />
    <p>
      Drag 'n' drop some files here, or click to select files
    </p>
    <aside
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "marginTop": 16,
        }
      }
    />
  </div>
</Fragment>
`;
